File: 2020/day1.py

with open('inputs/day1.txt', 'r') as f:
    numbers = list(map(int, f.read().split('\n')))

numbers_set = set(numbers)

# Part one
for n in numbers:
    if 2020 - n in numbers_set:
        print(f'Part one: {n * (2020 - n)}')
        break

# Part two
for i, a in enumerate(numbers):
    for j, b in enumerate(numbers[i+1:]):
        if 2020 - a - b in numbers_set:
            print(f'Part two: {a * b * (2020 - a - b)}')
            break

File: 2021/day1.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
)

func ParseNumbers(input string) []int {
	var numbers []int

	for _, number := range strings.Split(input, "\n") {
		iNumber, err := strconv.Atoi(number)
		if err != nil {
			panic(err)
		}

		numbers = append(numbers, iNumber)
	}
	return numbers
}

func CountAscending(numbers []int, windowSize int) int {
	var count int
	for i := windowSize; i < len(numbers); i++ {
		if numbers[i] > numbers[i-windowSize] {
			count++
		}
	}
	return count
}

func main() {
	input, err := ioutil.ReadFile("input/day1.txt")
	if err != nil {
		panic(err)
	}

	numbers := ParseNumbers(string(input))
	fmt.Printf("Day 1, part 1: %d\n", CountAscending(numbers, 1))
	fmt.Printf("Day 1, part 1: %d\n", CountAscending(numbers, 3))
}
